### ISSUE TITLE
`$unflatten` does not allow for deserialization of vectors

### ISSUE THREAD

**Issue by benpye** [tafia/quick-xml#463]
I have some input XML that looks like

```rust
<outer>
    <extra>Hello World</extra>
    <array>
        <entry test="x"></entry>
        <entry test="y"></entry>
    </array>
</outer>
```

I want to produce something like:

```
#[derive(Debug, Deserialize, PartialEq)]
struct Entry {
    test: String,
}

#[derive(Debug, Deserialize, PartialEq)]
struct Outer {
    extra: String,
    #[serde(rename = "$unflatten=entry")]
    array: Vec<Entry>,
}
```

Unfortunately this doesn't seem to work - though it isn't entirely surprising as the rename seems incompatible with the name of `array`.

**Comment by Mingun** [comment]
You shouldn't use `$unflatten` at all -- for deserializing it doesn't do anything, and for a serializing it's a poor way to force serialization as elements instead of attributes. For the last you can try to use [a newtype idiom](https://github.com/tafia/quick-xml/issues/280#issuecomment-1198871958).

Your issue is the same as #390. In the nutshell -- is is impossible to directly deserialize container with vector fields into the vector -- you should define an intermediate struct:
```rust
#[derive(Debug, Deserialize, PartialEq)]
struct Entry {
    test: String,
}

#[derive(Debug, Deserialize, PartialEq)]
struct Outer {
    extra: String,
    array: Intermediate,
}

#[derive(Debug, Deserialize, PartialEq)]
struct Intermediate {
    entry: Vec<Entry>,
}
```

Closed as duplicate of #390

**Comment by benpye** [comment]
Ah I see - I didn't realise that `$unflatten` had no effect on deserialization. I was hoping to avoid the intermediate struct, but that's what I ended up with and it does work, it's just slightly less ergonomic.
